apps/api/src/music/music.module.spec.ts

```diff
@@ -38,131 +37,179 @@ describe('Music Module', () => {
+        );
+
       expect(res.status).toBe(201);
-      expect(res.body.message).toBe('Music was successfully created');
+      expect(res.body.data.musicAudioFileName).toBeDefined();
     });
   });
 
-  describe('GET /api/v1/music/file/:id | Get music file', () => {
-    it('should find the music and get the file to play', async () => {
+  describe('Upload and get Music Cover Image File', () => {
+    it('GET /api/music/:id/cover | should find the music and get the image cover file', async () => {
       musicService.removeAllMusic();
 
-      const resForSetup = await request(app.getHttpServer())
+      const resForCreation = await request(app.getHttpServer())
         .post('/music')
-        .attach('file', 'assets/mock-data/audio-file.mp3')
-        .field('name', 'Music');
-
-      const musicId = resForSetup.body.data.id;
+        .send({ name: 'Music', releaseDate: new Date(Date.now()) });
+      const musicId = resForCreation.body.data.id;
+      await request(app.getHttpServer())
+        .post(`/music/${musicId}/cover/upload`)
+        .attach(
+          'coverImageFileName',
+          'assets/mock-data/musics-cover/android-chrome-512x512.png'
+        );
 
       const res = await request(app.getHttpServer()).get(
-        `/music/file/${musicId}`
+        `/music/${musicId}/cover`
       );
       expect(res.body.stream.path).toBeDefined();
     });
+
+    it('POST /api/music/:id/upload | should find the music and upload the file successfully', async () => {
+      musicService.removeAllMusic();
+
+      const resForCreation = await request(app.getHttpServer())
+        .post('/music')
+        .send({ name: 'Music', releaseDate: new Date(Date.now()) });
+      const musicId = resForCreation.body.data.id;
+      const res = await request(app.getHttpServer())
+        .post(`/music/${musicId}/cover/upload`)
+        .attach(
+          'coverImageFileName',
+          'assets/mock-data/musics-cover/android-chrome-512x512.png'
+        );
+
+      expect(res.status).toBe(201);
+      expect(res.body.data.coverImageFileName).toBeDefined();
+    });
   });
 
-  describe('GET /api/v1/music | Find and get all musics', () => {
+  describe('GET /api/music | Find and get all musics', () => {
     it('should return 3 music', async () => {
       await musicService.removeAllMusic();
       await musicService.create({
         name: 'Music #1',
-        fileName: 'Music #1 file path',
         releaseDate: new Date(Date.now()),
+        coverImageFileName: 'fa123asd.png',
+        musicAudioFileName: 'fa123asd.mp3',
       });
       await musicService.create({
         name: 'Music #2',
-        fileName: 'Music #2 file path',
-        releaseDate: new Date(Date.now()),
-      });
-      await musicService.create({
-        name: 'Music #3',
-        fileName: 'Music #3 file path',
         releaseDate: new Date(Date.now()),
+        coverImageFileName: 'asdfasd.png',
+        musicAudioFileName: 'asdfasd.mp3',
```

Comment:
Ewe kê çepokit be Keyboard -î da dawe lêre? :P

Reply:
Ewende qerizdarîm bo xoşîm nazanîm çidekem :D